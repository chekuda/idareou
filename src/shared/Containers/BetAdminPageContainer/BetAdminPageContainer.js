import React, { Component, Fragment } from 'react'
import { graphql } from 'react-apollo'
import { findBet } from '~src/shared/graphql/queries'
import BetAdminCard from '~src/shared/Components/BetAdminCard/BetAdminCard'

import Proptypes from 'prop-types'

export class BetAdminPageContainer extends Component {
  render() {
    const { data: { findBet: bet } } = this.props

    return <Fragment>
      <BetAdminCard { ...bet } />
    </Fragment>
  }
}

BetAdminPageContainer.propTypes = {
  betId: Proptypes.string.isRequired
}

export default graphql(findBet, { options: ({ betId }) => ({ variables: { id: betId } }) })(BetAdminPageContainer)
