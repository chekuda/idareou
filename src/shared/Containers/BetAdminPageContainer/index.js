
export { default as BetAdminPageContainer } from './BetAdminPageContainer'
