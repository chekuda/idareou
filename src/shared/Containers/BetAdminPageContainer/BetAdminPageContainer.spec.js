import React from 'react'
import { shallow } from 'enzyme'

import { BetAdminPageContainer } from './BetAdminPageContainer'

const mockBet = {
  title: 'mockTitle',
  description: 'mockDescription',
  amount: 'mockAmount',
  partecipants: ['mockPartecipant'],
  options: [{ _id: 'mockId', title: 'mockTitle' }, { _id: 'mockId', title: 'mockTitle' }]
}

const initProps = {
  data: { findBet: mockBet }
}

describe('BetAdminPageContainer', () => {
  it('renders without errors', () => {
    const component = shallow(<BetAdminPageContainer {...initProps } />)

    expect(component).toMatchSnapshot()
  })
  it('renders BetAdminCard', () => {
    const component = shallow(<BetAdminPageContainer {...initProps } />)

    expect(component.find('BetAdminCard').props()).toEqual(mockBet)
  })
})
