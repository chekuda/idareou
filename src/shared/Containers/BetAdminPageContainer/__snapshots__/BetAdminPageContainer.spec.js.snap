// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetAdminPageContainer renders without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BetAdminPageContainer
    data={
      Object {
        "findBet": Object {
          "amount": "mockAmount",
          "description": "mockDescription",
          "options": Array [
            Object {
              "_id": "mockId",
              "title": "mockTitle",
            },
            Object {
              "_id": "mockId",
              "title": "mockTitle",
            },
          ],
          "partecipants": Array [
            "mockPartecipant",
          ],
          "title": "mockTitle",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <BetAdminCard
        amount="mockAmount"
        description="mockDescription"
        options={
          Array [
            Object {
              "_id": "mockId",
              "title": "mockTitle",
            },
            Object {
              "_id": "mockId",
              "title": "mockTitle",
            },
          ]
        }
        partecipants={
          Array [
            "mockPartecipant",
          ]
        }
        title="mockTitle"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "amount": "mockAmount",
        "description": "mockDescription",
        "options": Array [
          Object {
            "_id": "mockId",
            "title": "mockTitle",
          },
          Object {
            "_id": "mockId",
            "title": "mockTitle",
          },
        ],
        "partecipants": Array [
          "mockPartecipant",
        ],
        "title": "mockTitle",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <BetAdminCard
          amount="mockAmount"
          description="mockDescription"
          options={
            Array [
              Object {
                "_id": "mockId",
                "title": "mockTitle",
              },
              Object {
                "_id": "mockId",
                "title": "mockTitle",
              },
            ]
          }
          partecipants={
            Array [
              "mockPartecipant",
            ]
          }
          title="mockTitle"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": "mockAmount",
          "description": "mockDescription",
          "options": Array [
            Object {
              "_id": "mockId",
              "title": "mockTitle",
            },
            Object {
              "_id": "mockId",
              "title": "mockTitle",
            },
          ],
          "partecipants": Array [
            "mockPartecipant",
          ],
          "title": "mockTitle",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
